bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/handler/EpsonProjectorHandler.java

```diff
@@ -156,10 +156,12 @@ private void updateChannelState(Channel channel) {
             State state = queryDataFromDevice(epsonCommand);
 
             if (state != null) {
-                updateStatus(ThingStatus.ONLINE);
                 if (isLinked(channel.getUID())) {
                     updateState(channel.getUID(), state);
                 }
+                if (state != UnDefType.UNDEF && getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
```

Comment:
Generally I think the thing should go online prior to any status updates, but that might just be a preference since I'm not aware of any documentation dictating that it _must_ be done that way.

But you also don't need to check the existing thing status since that will automatically be checked by the `updateStatus` implementation.
```suggestion
                if (state != UnDefType.UNDEF) {
                    updateStatus(ThingStatus.ONLINE);
                }
```

Reply:
Fixed. Thanks!